File: src/main/java/kitchenpos/core/products/tobe/domain/support/DefaultProductNamePolicy.java
package kitchenpos.core.products.tobe.domain.support;

import kitchenpos.core.products.tobe.domain.ProductNamePolicy;
import kitchenpos.core.shared.domain.ProfanityChecker;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class DefaultProductNamePolicy implements ProductNamePolicy {

    private final ProfanityChecker profanityChecker;

    public DefaultProductNamePolicy(ProfanityChecker profanityChecker) {
        this.profanityChecker = profanityChecker;
    }

    @Override
    public ProductNameValidationResult validate(String name) {
        List<String> errors = new ArrayList<>();
        if (name == null || name.isBlank()) {
            errors.add("Product name은 null 이거나 빈 값이 될 수 없습니다.");
        }
        if (name != null && profanityChecker.containsProfanity(name)) {
            errors.add("Product name에 비속어가 포함될 수 없습니다.");
        }
        return new ProductNameValidationResult(errors.isEmpty(), errors);
    }
}

File: src/main/kotlin/kitchenpos/product/tobe/infra/DefaultProductNamePolicy.kt
package kitchenpos.product.tobe.infra

import kitchenpos.common.annotation.DomainService
import kitchenpos.common.domain.Profanities
import kitchenpos.product.tobe.domain.ProductNamePolicy

@DomainService
class DefaultProductNamePolicy(private val profanities: Profanities) : ProductNamePolicy {
    override fun validate(name: String) {
        require(name.isNotBlank()) { "상품 이름은 필수값입니다." }
        require(!profanities.contains(name)) { "상품 이름에 금지어가 포함되어 있습니다." }
    }
}
